README.md

```diff
@@ -174,3 +175,70 @@ Submit a Pull Request. Classes should have the following sections, in order:
+  * you should search google/stack overflow to the best of your ability
+  * you should draw pictures, which will help to clarify both pointer questions and questions about your design (in terms of interactiosn between modules)
+  * you should come up with an educated guess for what you believe the answer is, or might be, and why that is the case
+  * you should explain to the TA what you have tried already. "My code is broken" and "Why is this segfault happening?" are common statements that a TA hears when they first walk up to a group, but are not helpful places to begin.
+* You will be forced to read technical documentation and header (`.h`) files that will be provided to you.  If you run into code or jargon that you don't understand (especially `typedef` related constructs), you should follow the steps above and then talk to a TA if still confused.  Plowing through the confusion is rarely the right thing to do, and will cost you, in some cases, a significant amount of time.
+* As long as you try your best during the class, the points will work themselves out in the end. That's why we give vague grades like "Very good", "good", etc.  Don't worry too much about your grades, don't stay in Halligan into the early morning hours, and please stay healthy :)
+
+### Combining Material
+
+* Write test programs.  If you are curious "what would happen if..." or "why is my code doing xyz...", you should change the code, recompile it, and see what changes.
+* Learn effective Googling strategies.  For example, googling `site:stackoverflow.com sscanf` will limit your search results to Stack Overflow.
+* Look for interesting questions on Stack Overflow's "Related" and "Hot Network Questions" sidebars.  I have learned a lot of interesting pieces of C trivia by following those links.
+* Write your own assembler for UMASM (the assembly language you write in during the last assignment)
+* I once had somebody ask me an amazing question when I TA'd for COMP 15:
+  * "We learned in lecture that we use a 'class' to store and model state and behavior.  I understand how we store state, because that's just data that we can store in memory.  But how does a class store behavior?"
+  * Spend some time thinking about the answer to this question.  The answer reveals a lot of deeper, subtle points about the relationship between code and data.
+* Explore other low-level, systems level languages such as Rust.  A good exercise is to implement the Universal Machine in Rust.
+* Since the C standard library is so small, implement various data structures or other tools that would be useful to have in a language:
+  * Implement various collections (hash tables, sets, etc.)
```

Comment:
I'm going to leave these in.  I think that it's a good exercise in familiarizing yourself with C idioms.  You can get a good feel for things like void pointers with implementing these collections in C, whereas you might use templates (or just implement monomorphic collections) in 15.